org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/part/DependencyEditorViewerPart.java

```diff
@@ -0,0 +1,214 @@
+	 * part of model changes. Since the selection is also managed by the model, they
+	 * should be ignored.
+	 */
+	private boolean ignoreSelectionChanges;
+
+	public DependencyEditorViewerPart(Composite parent, TargetDependencyModel model) {
+		super(parent, new TableColumnLayout(), model);
+		createContent();
+	}
+
+	public TableColumnLayout getLayout() {
+		return (TableColumnLayout) layout;
+	}
+
+	private void createContent() {
+		createTable();
+		createTableColumns();
+		newKeyListener();
+		newSelectionChangeListener();
+		newModelChangeListener();
+		doResizeColumns();
+	}
+
+	private void createTable() {
+		viewer = new TableViewer(composite);
+		viewer.setContentProvider(ArrayContentProvider.getInstance());
+		viewer.getTable().setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		viewer.getTable().setHeaderVisible(true);
+	}
+
+	private void createTableColumns() {
+		Function<MavenTargetDependency, String> getter = MavenTargetDependency::getGroupId;
+		BiConsumer<MavenTargetDependency, String> setter = MavenTargetDependency::setGroupId;
+		TableViewerColumn tableColumn = newGenericColumn(250, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_1);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getArtifactId;
+		setter = MavenTargetDependency::setArtifactId;
+		tableColumn = newGenericColumn(250, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_2);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getVersion;
+		setter = MavenTargetDependency::setVersion;
+		tableColumn = newGenericColumn(250, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_3);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getClassifier;
+		setter = MavenTargetDependency::setClassifier;
+		tableColumn = newGenericColumn(125, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_4);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getType;
+		tableColumn = newGenericColumn(125, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_5);
+		tableColumn.setEditingSupport(new MavenTypeEditingSupport(viewer, model));
+	}
+
+	private TableViewerColumn newGenericColumn(int weight, Function<MavenTargetDependency, String> getter) {
+		TableViewerColumn tableColumn = new TableViewerColumn(viewer, SWT.NONE);
+		tableColumn.getColumn().addSelectionListener(new TableColumnSortListener(viewer, getter));
+		tableColumn.setLabelProvider(new MavenTargetColumnLabelProvider<>(getter));
+		getLayout().setColumnData(tableColumn.getColumn(), new ColumnWeightData(weight, false));
+		return tableColumn;
+	}
```

Comment:
All classes in the new package org.eclipse.m2e.pde.ui.target.editor.internal.viewer could be inlined with the following change. This would save many lines by removing boilerplate code and would IMHO make it simpler to understand because one does not have to navigate through multiple classes.

You can added the classes documentation as documention of the corresponding methods
```suggestion
	private void createTableColumns() {
		Function<MavenTargetDependency, String> getter = MavenTargetDependency::getGroupId;
		BiConsumer<MavenTargetDependency, String> setter = MavenTargetDependency::setGroupId;
		TableViewerColumn tableColumn = newGenericColumn(250, getter);
		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_1);
		tableColumn.setEditingSupport(createMavenTextEditingSupport(getter, setter));

		getter = MavenTargetDependency::getArtifactId;
		setter = MavenTargetDependency::setArtifactId;
		tableColumn = newGenericColumn(250, getter);
		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_2);
		tableColumn.setEditingSupport(createMavenTextEditingSupport(getter, setter));

		getter = MavenTargetDependency::getVersion;
		setter = MavenTargetDependency::setVersion;
		tableColumn = newGenericColumn(250, getter);
		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_3);
		tableColumn.setEditingSupport(createMavenTextEditingSupport(getter, setter));

		getter = MavenTargetDependency::getClassifier;
		setter = MavenTargetDependency::setClassifier;
		tableColumn = newGenericColumn(125, getter);
		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_4);
		tableColumn.setEditingSupport(createMavenTextEditingSupport(getter, setter));

		getter = MavenTargetDependency::getType;
		tableColumn = newGenericColumn(125, getter);
		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_5);
		tableColumn.setEditingSupport(createMavenTypeEditingSupport());
	}

	private EditingSupport createMavenTextEditingSupport(Function<MavenTargetDependency, String> getter,
			BiConsumer<MavenTargetDependency, String> setter) {
		return createMavenEditingSupport(getter, setter, TextCellEditor::new);
	}

	private static final String[] DEFAULT_TYPES = new String[] { "jar", "bundle", "pom" };

	@SuppressWarnings("restriction")
	private EditingSupport createMavenTypeEditingSupport() {
		return createMavenEditingSupport(MavenTargetDependency::getType, MavenTargetDependency::setType, c -> {
			var editor = new org.eclipse.m2e.core.ui.internal.components.TextComboBoxCellEditor(c, SWT.NONE);
			editor.setItems(DEFAULT_TYPES);
			return editor;
		});
	}

	private EditingSupport createMavenEditingSupport(Function<MavenTargetDependency, String> getter,
			BiConsumer<MavenTargetDependency, String> setter, Function<Composite, CellEditor> cellEditorFactory) {
		return new EditingSupport(viewer) {

			@Override
			protected boolean canEdit(Object element) {
				return element instanceof MavenTargetDependency;
			}

			@Override
			protected Object getValue(Object element) {
				return getter.apply((MavenTargetDependency) element);
			}

			@Override
			protected void setValue(Object element, Object value) {
				MavenTargetDependency dependency = (MavenTargetDependency) element;
				String oldValue = getter.apply(dependency);
				String newValue = (String) value;

				// The table column needs to be redrawn. Otherwise we still see the old value...
				BiConsumer<MavenTargetDependency, String> realSetter = (k, v) -> {
					setter.accept(k, v);
					getViewer().update(element, null);
				};

				model.modify(dependency, realSetter, oldValue, newValue);
			}

			@Override
			protected CellEditor getCellEditor(Object element) {
				return cellEditorFactory.apply((Composite) getViewer().getControl());
			}
		};
	}

	private TableViewerColumn newGenericColumn(int weight, Function<MavenTargetDependency, String> getter) {
		TableViewerColumn tableColumn = new TableViewerColumn(viewer, SWT.NONE);
		tableColumn.getColumn().addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
			final boolean sortAscending;
			if (viewer.getTable().getSortColumn() != e.widget) {
				// When we don't currently sort by this column, default to ascending order
				sortAscending = true;
			} else {
				// Switch between ascending & descending order
				sortAscending = viewer.getTable().getSortDirection() != SWT.DOWN;
			}
			viewer.getTable().setSortDirection(sortAscending ? SWT.DOWN : SWT.UP);
			viewer.getTable().setSortColumn((TableColumn) e.widget);

			viewer.setComparator(new ViewerComparator() {
				@Override
				public int compare(Viewer viewer, Object e1, Object e2) {
					String left = getter.apply((MavenTargetDependency) e1);
					String right = getter.apply((MavenTargetDependency) e2);
					return sortAscending ? left.compareTo(right) : right.compareTo(left);
				}
			});
			viewer.refresh();
		}));
		tableColumn.setLabelProvider(new MavenTargetColumnLabelProvider<>(getter));
		getLayout().setColumnData(tableColumn.getColumn(), new ColumnWeightData(weight, false));
		return tableColumn;
	}
```